### PR TITLE
Checkpointing: use more explicit types in arguments

### LINKED ISSUE #32: Checkpointing: use more explicit types in arguments
From the audit:

> APIs of the `Checkpointing` and `ActivePeriod` libraries can be more explicit about supported data types (`uint64` for time-like values and `uint192` for numeric values). We suggest using exact data types and forcing users of the libraries to acknowledge that by using type casts.
Interestingly enough, there is the ready-made `getBlockNumber64` function, which fits perfectly into the picture.

The original intention was to avoid users having to cast, saving both gas costs and implementation details, but on second though, providing a more explicit interface from the libraries is better.

### PR DESCRIPTION
**Pull request by bingen**
Fixes #32.

### CODE REVIEW

File: shared/contract-utils/contracts/interfaces/IERC20WithCheckpointing.sol [hunk 1]
```diff
@@ -8,6 +8,6 @@ import "@aragon/os/contracts/lib/token/ERC20.sol";
 
 
 contract IERC20WithCheckpointing is ERC20 {
-    function balanceOfAt(address _owner, uint256 _blockNumber) public view returns (uint256);
-    function totalSupplyAt(uint256 _blockNumber) public view returns (uint256);
+    function balanceOfAt(address _owner, uint64 _blockNumber) public view returns (uint256);
```
Comment (sohkai) [inline shared/contract-utils/contracts/interfaces/IERC20WithCheckpointing.sol]: Given that MiniMe uses `uint256` as its time value in public interfaces, we shouldn't change this interface to `uint64`.

File: shared/contract-utils/test/checkpointing.js [hunk 2]
```diff
@@ -96,27 +96,4 @@ contract('Checkpointing', () => {
 
     await assertRevert(checkpointing.addCheckpoint(time - 1, value))
   })
-
-  const UINT64_OVERFLOW = (new web3.BigNumber(2)).pow(new web3.BigNumber(64))
-  const UINT192_OVERFLOW = (new web3.BigNumber(2)).pow(new web3.BigNumber(192))
-
-  it('fails if set value is too high', async () => {
```
Comment (sohkai) [inline shared/contract-utils/test/checkpointing.js:101]: In line with https://github.com/aragonone/voting-connectors/pull/37#pullrequestreview-337786992, we should also make sure to double check in both the TokenWrapper and VotingAggregator that we're catching the casts correctly for any input values.

File: shared/contract-utils/contracts/CheckpointingHelpers.sol [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+
+library CheckpointingHelpers {
+    uint256 private constant MAX_UINT64 = uint64(-1);
+    uint256 private constant MAX_UINT192 = uint192(-1);
+
+    string private constant ERROR_UINT64_TOO_BIG = "UINT64_NUMBER_TOO_BIG";
+    string private constant ERROR_UINT192_TOO_BIG = "UINT192_NUMBER_TOO_BIG";
+
+    function toUint64Time(uint256 _a) internal pure returns (uint64) {
```
Comment (sohkai) [inline shared/contract-utils/contracts/CheckpointingHelpers.sol:11]: This comes at the expense of a bit more bytecode in the base contract, but avoids the nameclash with the already existing `Uint256Helpers.toUint64()`.

It's not really ideal... but it is what it is given our limitations with mixing libraries together.

Comment (bingen) [reply shared/contract-utils/contracts/CheckpointingHelpers.sol:11]: Yes, exactly, I guess I would have called the lib `Uint64Helpers` and then have another one for `uint192` and so on, as now modifying `aragonOS` one is not an agile process...

**Pull request merged**